### PR TITLE
Fluent Calendar

### PR DESCRIPTION
**Pull request by amwx**
## What does the pull request do?
Adds Fluent/WinUI themed Calendar. MS handles some of the styling for calendar day items in code so some styles are based on observed behavior. I didn't know how to test it with the Avalonia source, but it worked in my test project

image

## Checklist

- [ ] Added unit tests (if possible)?
- [ ] Added XML documentation to any related classes?
- [ ] Consider submitting a PR to https://github.com/AvaloniaUI/Avaloniaui.net with user documentation

### CODE REVIEW

**Comment by jmacato** [comment]
It looks like this here
image

**Comment by danwalmsley** [comment]
Doesn't look right, probably missing resources , I'll take a look.

**Comment by amwx** [comment]
I've updated the Button's to not have rounded corners and center the day numbers & fixed a couple color resources that were incorrect. I've also added the disabled style which I forgot. 
The transitions could be added, but there's a couple issues
1-Calendar doesn't currently have pseudoclasses for each display mode
2-WinUI has 3 separate containers, one for each display mode, Avalonia only has two (year/decade share), so currently transition effects could only work when going month -> year or year -> month
3-Another container could be added, but it would require a change in the default theme style most likely
I'm hesitant to touch the source code cause I don't want to break anything inadvertently as the Calendar is a complex beast

**Comment by danwalmsley** [comment]
> I've updated the Button's to not have rounded corners and center the day numbers & fixed a couple color resources that were incorrect. I've also added the disabled style which I forgot.
> The transitions could be added, but there's a couple issues
> 1-Calendar doesn't currently have pseudoclasses for each display mode
> 2-WinUI has 3 separate containers, one for each display mode, Avalonia only has two (year/decade share), so currently transition effects could only work when going month -> year or year -> month
> 3-Another container could be added, but it would require a change in the default theme style most likely
> I'm hesitant to touch the source code cause I don't want to break anything inadvertently as the Calendar is a complex beast

We dont need to update the default theme, as long as we dont break existing functionality...

default theme will be deprecated.

**Review by danwalmsley — APPROVED** [review]
LGTM, we will have to revisit to add the transitions.

File: src/Avalonia.Themes.Fluent/CalendarButton.xaml [hunk 1]
```diff
@@ -7,75 +7,115 @@
...
+    <Setter Property="Foreground" Value="{DynamicResource CalendarViewBlackoutForeground}"/>
   </Style>
-  <Style Selector="CalendarButton:btnfocused /template/ Rectangle#FocusVisual">
-    <Setter Property="IsVisible" Value="True"/>
+
+  <!--<Style Selector="CalendarButton:dayfocused /template/ Border#FocusVisual">
+            <Setter Property="IsVisible" Value="True"/>
+      </Style>-->
+
+  <Style Selector="CalendarDayButton:disabled /template/ ContentControl#Content">
```
Comment (grokys) [inline src/Avalonia.Themes.Fluent/CalendarButton.xaml:118]: Is this a mistake? Should it be `CalendarButton`?

**Pull request merged**